qt/applications/workbench/mantidworkbench.cpp

```diff
@@ -129,7 +144,16 @@ decltype(boost::this_process::environment()) childEnvironment(const fs::path &di
 int startWorkbench(const fs::path &dirOfExe, int argc, char **argv) {
   ExeArgs startupWorkbench{"-m", WORKBENCH_MAIN};
   appendArguments(&startupWorkbench, argc, argv);
-  return bp::system(pythonExecutable(dirOfExe), startupWorkbench, childEnvironment(dirOfExe));
+  try {
+    return bp::system(pythonExecutable(dirOfExe), startupWorkbench, childEnvironment(dirOfExe));
+  } catch (const bp::process_error &exc) {
+    std::cerr << "Running " << pythonExecutable(dirOfExe) << " ";
+    for (const auto &arg : startupWorkbench) {
+      std::cerr << arg << " ";
+    }
+    std::cerr << "\nCaught system_error with code " << exc.code() << " meaning " << exc.what() << "\n";
+    return 1;
```

Comment:
Return the error code, not 1
On POSIX systems the return code can be significant (including if it is negative) and we effectively discard this here

Reply:
Couldn't agree more!